### DISCUSSION TITLE
set pickable on deck instance

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#7615] [Q&A]
Hi, I'm curious if it's possible to set globally `pickable` property on deck instance (currently it works only for false value as I understand it from docs) for all the layers without constructing each layer with `pickable: true`?

I'm using `pickObject()` in some cases (like measuring) and I need Z from 3D layers. It would be easier for me to set `pickable: true` on deck instance which overrides `pickable` on layers. When false, all the layers will have `pickable` as constructed.

Basically in normal mode I have few pickable layers, but when I turn on measuring, I need all the layers to be pickable, ideally without constructing layers again (I separated methods for building layers and it means I need to pass another parameter to them which increases complexity)

Thanks

**Comment by Pessimistress** [comment]
The general strategy for this kind of "sometimes pickable" use case is to set `pickable: true` on all layers, then provide a [layerFilter](https://deck.gl/docs/api-reference/core/deck#layerfilter):

```js
function layerFilter({layer, renderPass}) {
	if (renderPass === 'picking:hover') {
		// the default picking pass
		// only returns true for a few layers
	}
	return true; 
}
```
